### DISCUSSION TITLE
Javafx issue

### DISCUSSION THREAD

**Discussion by Super-JK** [LemaireEmilien/calculator-cucumber-2024#21] [General]
@Faltmatix Je propose on contine la discussion ici si il y a encore des problemes.

Selon ton erreur java 17 (version 61) lance le code. Cela peut etre un prblm si ton JAVA_HOME est mal configuré.
J'ai aussi update le pom.xml pour que le jar lance bien al gui. La ici il faut bien verifier que ca se lance avec java 21.

Dans le pire des cas supprime java 17 du PATH ou meme du pc. Ou alors utilise une vm/container.

**Comment by Super-JK** [comment]
Nico a tester et il y a un prblm avec intelij sur windows il ne prend pas le bon jre. 
Si tu utilise bien a java 21 dans un terminal (java --version) cela fonctionne

**Comment by Faltmatix** [comment]
En effet, il y avait bien le problème intellij + windows. Bien vu, javafx fonctionne maintenant. Je mets ici les étapes que j'ai faites pour résoudre le problème de mon côté:

- S'assurer que le java windows par défaut soit >= 21.
- Run `mvn clean install compile`
- Run `mvn jar:jar`
- Exécuter le jar avec `java -jar $CHEMIN_DU_JAR`

C'est pas idéal qu'intellij n'arrive pas à run les tâches. Ca rend le dev sur Windows un peu chaotique. J'investiguerai le problème demain et si j'ai du nouveau je ferai une update ici.

## Update

Une fois que le chemin de java est bien setup avec les variables d'environnements Windows, on peut modifier le runner intellij pour maven pour qu'il prenne ce java_home au lieu du jdk par défaut.

![image](https://github.com/LemaireEmilien/calculator-cucumber-2024/assets/49649047/b70b3e7b-64b8-4693-a42e-8ad7264990f7)

**Comment by Super-JK** [comment]
@Faltmatix tu as encore des modifs a faire ou on peut mettre sur main ?
